### ISSUE TITLE
Getting information about the target element

### ISSUE THREAD

**Issue by emilohlund-git** [isaacHagoel/svelte-dnd-action#462]
I've been scouring the documentation but I can't seem to find what I'm looking for.

If I have, lets say a hand of cards, and I want to match two of them. I would like to get information, such as the data attribute and ID of the target element onto which I dragged the source element into. 

Is there any built in way to do this?

**Comment by isaacHagoel** [comment]
If by "target element" you mean "target zone" then you know which one
trivially because the handler that got triggered is the one that belongs to
it and it has access to its properties (e.g. id). Does that make sense?

**Comment by emilohlund-git** [comment]
Yeah, I got you. My use-case is probably a little bit different then what the library is intended for. I'm using 1 zone, which is the player's 'hand'. 

Thank you for your quick reply! :)

**Comment by isaacHagoel** [comment]
One thing you could try is to listen on the library's internal events which might have what you need.
See https://github.com/isaacHagoel/svelte-dnd-action/blob/4f90e31e175bbc807d6b2ca628adc9475e63c2e1/src/helpers/dispatcher.js#L36

**Comment by McCrafterIV** [comment]
I face the same problem. When an element is dragged from zone A to zone B, I need to be notified on zone A (e.g. via an event listener) that the element was dragged to zone B instead of any other zone. Listening to the from you mentioned events didn't work for me. Could you proved a small code snippet?

**Comment by isaacHagoel** [comment]
Have you tried using the information included in the consider events?
What's missing?

**Comment by McCrafterIV** [comment]
The consider event listening on zone A only gives the zone A HTML elements in the event details. Nowhere is Zone B mentioned. `originalTarget`, `explicitOriginalTarget` and `srcElement` are the same - zone A.
A property that references zone B is missing.

**Comment by isaacHagoel** [comment]
Sure but you can use a simple store and update it every time a drag start
in any zone. Would that not work (that's how it is usually done)?
Currently the library does not require the zones to provide any IDs (unlike
the items who do have IDs) therefore it can't include the info in the event

**Comment by McCrafterIV** [comment]
If I create a global store which saves the current element on consider events, it will update it each time I drag the element over a new zone.
For example: If I drag the element from zone A over zone C to zone B, the best Information I could get with this would be zone C (zone B we could maybe filter out..), as consider is triggered while dragging over a zone even if we don't drop it there. finalize doesn't work here (at least I couldn't find a solution using finalize yet), because the finalize of zone B may be triggered before the finalize of zone A, which results in the store not being updated.

**Comment by isaacHagoel** [comment]
Not the current element being dragged. I thought you wanted to know the
origin zone. Each consider events has information about the 'trigger' of
the event. If it says that drag started you update the store, on finalize
you update it again (to undefined)

**Comment by McCrafterIV** [comment]
I completely missed the trigger parameter, thank you! Using the trigger parameter I can implement your suggestion. This solves my problem for now.
Thank you for the help and awesome library!

**Comment by isaacHagoel** [comment]
Happy to help. Let me know if you need anything else

**Comment by tomconnors** [comment]
I believe I have an issue similar to @emilohlund-git's. I only have one dnd-zone. In on:consider events, I'd like to know what element the mouse is over. Other DND libraries refer to this as the drop target. It doesn't appear that this information is available in the on:consider event details.

**Comment by isaacHagoel** [comment]
What is the problem you are trying to solve?
If you describe it I might be able to offer a solution

**Comment by tomconnors** [comment]
I am trying to make a UI that allows items to be grouped and sorted. 

Initially, we just have a list of items:
`[{id: 1, label: "Item 1"}, {id: 2, label: "Item 2"}, {id: 3, label: "Item 3"}]`.

If the user drags any of these items on top of another item, they should form a group. So if I drag item 1 on top of item 2, I should get:
`[{id: 4, items: [{id: 1, label: "Item 1"}, {id: 2, label: "Item 2"}]}, {id: 3, label: "Item 3"}]`.

If I then continue dragging item 1 past item 2, but not out of the group's DOM area, item 1 should be after item 2 in the group:
`[{id: 4, items: [ {id: 2, label: "Item 2"}, {id: 1, label: "Item 1"}]}, {id: 3, label: "Item 3"}]`.

If I then continue dragging item 1 past the group's boundary, but not quite to item 3, the group should be removed, and item 1 should now come after item 2 in the flat list of items:
`[{id: 2, label: "Item 2"}, {id: 1, label: "Item 1"}, {id: 3, label: "Item 3"}]`.

If I recreate that group, then drag item 3 into it, I should end up with one group containing all three items: 
`[{id: 4, items: [{id: 1, label: "Item 1"}, {id: 2, label: "Item 2"}], {id: 3, label: "Item 3"}}]`.

I'd like to know what the drop target is so that I can create the groups containing the dragging item and the hovered item.

**Comment by isaacHagoel** [comment]
What is the difference between this and the crazy nesting example?

**Comment by tomconnors** [comment]
The crazy nesting example does not create groups based on drag operations. It just reorders existing groups and items.

**Comment by isaacHagoel** [comment]
I see. I am out camping
Will block time to look at this at the beginning of the week

**Comment by tomconnors** [comment]
Thanks for taking the time to respond. Enjoy your weekend.

**Comment by isaacHagoel** [comment]
Okay, I read it again - more throughly this time.
Within this lib when you drag an item over another it means that you are trying to place the dragged item instead of the item that it is dragged over. 
With that said, maybe it is possible to hack it somehow to achieve what you're after using the library's internal events (you should be able to listen on them). See [here](https://github.com/isaacHagoel/svelte-dnd-action/blob/8f8b80fb69fbf0ab07eeb0ab0dde5debf68d68cc/src/helpers/dispatcher.js#L36).

**Comment by isaacHagoel** [comment]
Closing because the question has been answered or a workable solution is documented in the discussion, and no further library change is currently outstanding. If this still affects v0.9.77, please open a focused issue with a current reproduction.
